Doc/c-api/long.rst

```diff
@@ -529,10 +529,190 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
+
+   *ndigits* is the number of digits in the *digits* array.
+
+   *digits* is an array of unsigned digits. Digits must be in the range
+   [``0``; ``PyLong_BASE - 1``].
+
+   See :c:struct:`PyLong_LAYOUT` for the layout of an array of digits.
+
+   See also the :ref:`PyLongWriter API <pylongwriter>`.
+
+
+.. c:struct:: PyLong_DigitArray
+
+   A Python :class:`int` object exported as an array of digits.
+
+   See :c:struct:`PyLong_LAYOUT` for the :c:member:`digits` layout.
+
+   .. c:member:: PyLongObject *obj
+
+      Strong reference to the Python :class:`int` object.
+
+   .. c:member:: int negative
+
+      1 if the number is negative, 0 otherwise.
+
+   .. c:member:: size_t ndigits
+
+      Number of digits in :c:member:`digits` array.
+
+   .. c:member:: Py_digit *digits
+
+      Array of unsigned digits.
+
+
+.. c:function:: int PyLong_Export(PyObject *obj, PyLong_DigitArray *array)
+
+   Export a Python :class:`int` object as an array of digits.
+
+   On success, set *\*array* and return 0.
+   On error, set an exception and return -1.
+
+   This function always succeeds if *obj* is a Python :class:`int` object or a
+   subclass.
+
+   :c:func:`PyLong_ReleaseExport` must be called once done with using
+   *export*.
+
+
+.. c:function:: void PyLong_ReleaseExport(PyLong_DigitArray *array)
+
+   Release the export *array* created by :c:func:`PyLong_Export`.
+
+
+.. _pylongwriter:
+
+PyLongWriter API
+^^^^^^^^^^^^^^^^
+
+The :c:func:`PyLong_Import` function can be used to create a Python
+:class:`int` from an array of digits. When converting an integer from a format
+to a Python :class:`int`, creating a temporary array of digits can be
+inefficient. In this case, the :c:type:`PyLongWriter` API can be used for even
+more efficient import.
+
+.. versionadded:: 3.14
+
+.. c:struct:: PyLongWriter
+
+   A Python :class:`int` writer instance.
+
+   The instance must be destroyed by :c:func:`PyLongWriter_Finish`.
+
+
+.. c:function:: PyLongWriter* PyLongWriter_Create(int negative, size_t ndigits, Py_digit **digits)
+
+   Create a :c:type:`PyLongWriter`.
+
+   On success, set *\*digits* and return a writer.
+   On error, set an exception and return ``NULL``.
+
+   *negative* is ``1`` if the number is negative, or ``0`` otherwise.
+
+   *ndigits* is the number of digits in the *digits* array.
+
+   The caller must initialize the array of digits *digits* and then call
+   :c:func:`PyLongWriter_Finish` to get a Python :class:`int`. Digits must be
+   in the range [``0``; ``PyLong_BASE - 1``].
+
+   See :c:struct:`PyLong_LAYOUT` for the layout of an array of digits.
+
+   See also the :c:func:`PyLong_Import` function.
+
+
+.. c:function:: PyObject* PyLongWriter_Finish(PyLongWriter *writer)
+
+   Finish a :c:type:`PyLongWriter` created by :c:func:`PyLongWriter_Create`.
+
+   On success, return a Python :class:`int` object.
+   On error, set an exception and return ``NULL``.
+
+Example with a single digit::
```

Comment:
It might worth to keep this example while discussing the API but, honestly, I find documentation clear enough without.

Reply:
Ok, I removed the example.